contracts/L2/token/LivepeerToken.sol

```diff
@@ -0,0 +1,45 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {ERC20, ERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+import {ILivepeerToken} from "./ILivepeerToken.sol";
+
+contract LivepeerToken is ILivepeerToken, AccessControl, ERC20Permit {
+    bytes32 public constant MINT_CONTROLLER = keccak256("MINT_CONTROLLER");
+
+    constructor() ERC20("Livepeer Token", "LPT") ERC20Permit("Livepeer Token") {
+        _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        _setRoleAdmin(MINT_CONTROLLER, DEFAULT_ADMIN_ROLE);
+    }
+
+    function addMintController(address _newController) external {
```

Comment:
Do we need this function? Could the same functionality be achieved if the address with the admin role made the following call?

```solidity
grantRole(MINTER_ROLE, account)
```

Reply:
yeah agreed, can provide hash of the role externally